barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -781,21 +784,16 @@ typename cycle_group<Builder>::batch_mul_internal_output cycle_group<Builder>::_
         }
     }
 
-    size_t num_bits = 0;
+    size_t num_bits = scalars[0].num_bits();
     for (auto& s : scalars) {
-        num_bits = std::max(num_bits, s.num_bits());
+        BB_ASSERT_EQ(s.num_bits(), num_bits, "Scalars of different bit-lengths not supported!");
     }
-    size_t num_rounds = (num_bits + TABLE_BITS - 1) / TABLE_BITS;
-
-    const size_t num_points = scalars.size();
+    size_t num_rounds = numeric::ceil_div(num_bits, TABLE_BITS);
 
     std::vector<straus_scalar_slice> scalar_slices;
     scalar_slices.reserve(num_points);
-    for (size_t i = 0; i < num_points; ++i) {
-        scalar_slices.emplace_back(straus_scalar_slice(context, scalars[i], TABLE_BITS));
```

Comment:
The purpose of `emplace_back` is to pass the constructor arguments directly so that the object can be constructed in place rather than constructed then moved. Its possible this would be optimized to be equivalent anyway but at the very least its poor code.